proto/trees.proto

```diff
@@ -0,0 +1,25 @@
+syntax = "proto3";
+
+import "hash.proto";
+
+// A key value pair used to define leaves of sparse merkle tree.
+message Item {
+    // When assigning this data structure into a Sparse Merkle Tree, only the
+    // first 64bits are used as a key.
+    Digest key = 1;
+    Digest value = 2;
+}
+
+// The data of a merkle mapping, to be materialzied as a sparse merkle tree.
+//
+// Note: this representation does not support partial view.
+message MerkleMap {
+    repeated Item items = 1;
+}
+
+// The data of a merkle tree, to be materialzied as a simple merkle tree.
+//
+// Note: this representation does not support partial view.
+message MerkleTree {
+    repeated Digest leaves = 1;
+}
```

Comment:
I think we'll need to iterate on this a bit. We have different data structures which end up being represented as various types of Merkle trees:

- Key-value maps with 64-bit keys (usually represented as Sparse Merkle trees). This would be probably be serialized simply as `(u64, Digest)`.
- Key-value maps with 256-bit keys (usually represented as Tiered Sparse Merkle trees). This would probably be serialized as `(Digest, Digest)`.
- Merkle store - can be thought of as a collection of Merkle trees - not sure what's the best way to serialize it yet.
- MMR - not sure what's the best way to serialize it yet.
- Maybe something else.